shell.nix

```diff
@@ -0,0 +1,53 @@
+{ pkgs ? import <nixpkgs> {} }:
+	let
+	overrides = (builtins.fromTOML (builtins.readFile ./rust-toolchain.toml));
+	libPath = with pkgs; lib.makeLibraryPath [
+		# load external libraries that you need in your rust project here
+	];
+in
+pkgs.mkShell rec {
+	# Kanidm dependencies
+	buildInputs = with pkgs; [
+		pkg-config
+		
+		git
+		rustup
+
+		clang
+		llvmPackages.bintools
+		
+		openssl
+		systemd
+		linux-pam
+	];
+	
+	RUSTC_VERSION = overrides.toolchain.channel;
+	# https://github.com/rust-lang/rust-bindgen#environment-variables
+	LIBCLANG_PATH = pkgs.lib.makeLibraryPath [ pkgs.llvmPackages_latest.libclang.lib ];
+	shellHook = ''
+		export PATH=$PATH:''${CARGO_HOME:-~/.cargo}/bin
+		export PATH=$PATH:''${RUSTUP_HOME:-~/.rustup}/toolchains/$RUSTC_VERSION-x86_64-unknown-linux-gnu/bin/
+		'';
+
+	# Add precompiled library to rustc search path
+	RUSTFLAGS = (builtins.map (a: ''-L ${a}/lib'') [
+		# add libraries here (e.g. pkgs.libvmi)
+	]);
+	LD_LIBRARY_PATH = libPath;
+	PKG_CONFIG_PATH = "${pkgs.openssl.dev}/lib/pkgconfig";
```

Comment:
You shouldn't and shouldn't need to override PKG_CONFIG_PATH like that. You might need to add openssl.dev to buildInputs instead if just openssl doesn't work

Reply:
I'll give that a go but when I was trying to get this shell to work it started working after including this ENV variable so I'll have to test without it to make sure it was unrelated.

Reply:
```
nix-repl> :p builtins.readFile (runCommand "" {buildInputs = [pkg-config openssl];} "exec > $out; echo $PKG_CONFIG_PATH_FOR_TARGET; pkg-config --libs openssl")
/nix/store/y940g0f3as2crh7frkb4p9f99wxzyzkm-openssl-3.3.2-dev/lib/pkgconfig
-L/nix/store/1a4j0c5xn6xl53z77sifkncg3fing1qj-openssl-3.3.2/lib -lssl -lcrypto
```

It should be setting PKG_CONFIG_PATH_FOR_TARGET (see https://github.com/NixOS/nixpkgs/issues/64530#issuecomment-1030313034).
If it's not working, what is reading this env var if not pkg-config that isn't respecting this?